cli/actions/importKeys_test.go

```diff
@@ -0,0 +1,295 @@
+
+	validatorCtId := "validatorctid"
+	validatorImportCtName := "validator-import-client"
+	validatorImportCtId := "validator-import-ct-id"
+
+	// Mock ContainerList
+	dockerClient.EXPECT().
+		ContainerList(gomock.Any(), types.ContainerListOptions{
+			All:     true,
+			Filters: filters.NewArgs(filters.Arg("name", services.ServiceCtValidator)),
+		}).
+		Return([]types.Container{
+			{ID: validatorCtId},
+		}, nil)
+	// Mock ContainerInspect
+	dockerClient.EXPECT().
+		ContainerInspect(gomock.Any(), services.ServiceCtValidator).
+		Return(types.ContainerJSON{
+			ContainerJSONBase: &types.ContainerJSONBase{
+				ID: validatorCtId,
+				State: &types.ContainerState{
+					Running: true,
+				},
+			},
+		}, nil).
+		Times(3)
+	// Mock ContainerStop
+	dockerClient.EXPECT().
+		ContainerStop(gomock.Any(), validatorCtId, gomock.Any()).
+		Return(nil)
+	// Mock ContainerCreate
+	dockerClient.EXPECT().
+		ContainerCreate(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), validatorImportCtName).
+		Return(container.ContainerCreateCreatedBody{ID: validatorImportCtId}, nil).
+		Times(1)
+	// Mock ContainerStart
+	dockerClient.EXPECT().
+		ContainerStart(gomock.Any(), validatorImportCtId, gomock.Any()).
+		Return(nil).
+		Times(1)
+	dockerClient.EXPECT().
+		ContainerStart(gomock.Any(), services.ServiceCtValidator, gomock.Any()).
+		Return(nil).
+		Times(1)
+	// Mock ContainerWait
+	exitCh := make(chan container.ContainerWaitOKBody, 1)
+	exitCh <- container.ContainerWaitOKBody{
+		StatusCode: 0,
+	}
+	dockerClient.EXPECT().
+		ContainerWait(gomock.Any(), validatorImportCtName, container.WaitConditionNextExit).
+		Return(exitCh, make(chan error)).
+		Times(1)
+	// Mock ContainerRemove
+	dockerClient.EXPECT().
+		ContainerRemove(gomock.Any(), validatorImportCtId, types.ContainerRemoveOptions{}).
+		Return(nil).
+		Times(1)
+
+	return dockerClient
+}
+
+// importKeysGoldenPatCustomImage returns a mocked docker client interface with
+// all the required responses for a correct custom validator import keys container execution.
+func importKeysGoldenPatCustomImage(t *testing.T, ctrl *gomock.Controller) client.APIClient {
+	dockerClient := mock_client.NewMockAPIClient(ctrl)
+
+	validatorCtId := "validatorctid"
+	validatorImportCtName := "validator-import-client"
+	validatorImportCtId := "validator-import-ct-id"
+
+	// Mock ContainerList
+	dockerClient.EXPECT().
+		ContainerList(gomock.Any(), types.ContainerListOptions{
+			All:     true,
+			Filters: filters.NewArgs(filters.Arg("name", services.ServiceCtValidator)),
+		}).
+		Return([]types.Container{
+			{ID: validatorCtId},
+		}, nil)
+	// Mock ContainerInspect
+	dockerClient.EXPECT().
+		ContainerInspect(gomock.Any(), services.ServiceCtValidator).
+		Return(types.ContainerJSON{
+			ContainerJSONBase: &types.ContainerJSONBase{
+				ID: validatorCtId,
+				State: &types.ContainerState{
+					Running: true,
+				},
+			},
+		}, nil).
+		Times(2)
```

Comment:
Is this `Times(...)` the only difference between the two Golden Paths?

Reply:
Refactored here d0455d80ddc06282139dc19750fc3e8eb0c88465